src/repository/db/db_repository.go

```diff
@@ -0,0 +1,260 @@
+	file, err := os.Create("./resources/characters.csv")
+
+	// ignoring close error it's safe on this point: https://www.joeshaw.org/dont-defer-close-on-writable-files/
+	defer file.Close()
+	defer readCharactersFromCSV()
+
+	if err != nil {
+		return _errors.NewInternalServerError(errorWritingFile)
+	}
+	writer := csv.NewWriter(file)
+	defer writer.Flush()
+
+	for _, ch := range characters {
+		var row []string
+		row = append(row, strconv.Itoa(ch.Id))
+		row = append(row, ch.Name)
+		row = append(row, ch.Status)
+		row = append(row, ch.Species)
+		row = append(row, ch.Type)
+		row = append(row, ch.Gender)
+		row = append(row, ch.Origin.Name)
+		row = append(row, ch.Origin.Url)
+		row = append(row, ch.Location.Name)
+		row = append(row, ch.Location.Url)
+		row = append(row, ch.Image)
+
+		var episodesString string
+		for _, episode := range ch.Episodes {
+			episodesString += fmt.Sprintf("+%s", episode)
+		}
+
+		row = append(row, episodesString)
+		if err := writer.Write(row); err != nil {
+			return _errors.NewInternalServerError(errorWritingFile)
+		}
+	}
+	return nil
+}
+
+//Get character from map (complex O(1) )
+func (db *dbRepository) GetCharacterFromId(id string) (*model.Character, _errors.RestError) {
+	if !isCsvFetched {
+		return nil, _errors.NewInternalServerError(errorDbEmpty)
+	}
+
+	ch, ok := charactersMap[id]
+	if ch == nil || !ok {
+		return nil, _errors.NewInternalServerError(fmt.Sprintf("character with id %s not found, fetch with more pages to update the db", id))
+	}
+
+	return ch, nil
+}
+
+//Get all characters from map
+func (db *dbRepository) GetCharacters() ([]model.Character, _errors.RestError) {
+	if !isCsvFetched {
+		return nil, _errors.NewInternalServerError(errorDbEmpty)
+	}
+
+	var characters []model.Character
+	for _, ch := range charactersMap {
+		characters = append(characters, *ch)
+	}
+
+	return characters, nil
+}
+
+//Get character id from csv map (complex O(n) )
+func (db *dbRepository) GetCharacterIdByName(name string) (string, _errors.RestError) {
+	if !isCsvFetched {
+		return "", _errors.NewInternalServerError(errorDbEmpty)
+	}
+
+	file, err := os.Open(viper.GetString("db.mapPath"))
+	defer file.Close()
+
+	if err != nil {
+		return "", _errors.NewInternalServerError(errorDbEmpty)
+	}
+
+	r := csv.NewReader(file)
+	for {
+		record, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return "", _errors.NewInternalServerError(errorWritingFile)
+		}
+
+		if strings.TrimSpace(strings.ToLower(record[1])) == strings.TrimSpace(strings.ToLower(name)) {
+			return record[0], nil
+		}
+	}
+
+	return "", _errors.NewNotFoundError(fmt.Sprintf("character with name %s not found", name))
+
+}
+
+func readCharactersFromCSV() bool {
+	// empty map
+	charactersMap = make(map[string]*model.Character)
+
+	file, err := os.Open(viper.GetString("db.charactersPath"))
+	defer file.Close()
+
+	if err != nil {
+		isCsvFetched = false
```

Comment:
Here `isCsvFetched` always be `false`
Why not return `isCsvFetched` instead of `false`?